vms/platformvm/api/camino_test.go

```diff
@@ -0,0 +1,434 @@
+		},
+		"Non Existing Deposit Offer": {
+			args: BuildGenesisArgs{
+				AvaxAssetID: ids.ID{},
+				NetworkID:   0,
+				UTXOs: []UTXO{
+					{
+						Address: addr,
+						Amount:  10,
+					},
+				},
+				Validators: []PermissionlessValidator{
+					{
+						Staker: Staker{
+							StartTime: 0,
+							EndTime:   20,
+							NodeID:    nodeID,
+						},
+						RewardOwner: &Owner{
+							Threshold: 1,
+							Addresses: []string{addr},
+						},
+						Staked: []UTXO{{
+							Amount:  weight,
+							Address: addr,
+						}},
+					},
+				},
+				Chains: []Chain{
+					{
+						VMID:     ids.GenerateTestID(),
+						SubnetID: ids.GenerateTestID(),
+					},
+				},
+				Camino: Camino{
+					VerifyNodeSignature: true,
+					LockModeBondDeposit: true,
+					InitialAdmin:        ids.ShortID{},
+					AddressStates:       nil,
+					DepositOffers:       nil,
+					ValidatorConsortiumMembers: []ids.ShortID{
+						ids.GenerateTestShortID(),
+					},
+					ValidatorDeposits: [][]UTXODeposit{
+						{
+							{
+								ids.GenerateTestID(),
+								10,
+								10,
+								"",
+							},
+						},
+					},
+					UTXODeposits: []UTXODeposit{
+						{
+							ids.GenerateTestID(),
+							10,
+							10,
+							"",
+						},
+					},
+				},
+			},
+			reply:       BuildGenesisReply{},
+			expectedErr: errNonExistingOffer,
+		},
+		"UTXO Has No Value": {
+			args: BuildGenesisArgs{
+				AvaxAssetID: ids.ID{},
+				NetworkID:   0,
+				UTXOs: []UTXO{
+					{
+						Address: addr,
+						Amount:  0,
+					},
+				},
+				Validators: []PermissionlessValidator{
+					{
+						Staker: Staker{
+							StartTime: 0,
+							EndTime:   20,
+							NodeID:    nodeID,
+						},
+						RewardOwner: &Owner{
+							Threshold: 1,
+							Addresses: []string{addr},
+						},
+						Staked: []UTXO{{
+							Amount:  weight,
+							Address: addr,
+						}},
+					},
+				},
+				Chains: []Chain{},
+				Camino: Camino{
+					VerifyNodeSignature: true,
+					LockModeBondDeposit: true,
+					ValidatorConsortiumMembers: []ids.ShortID{
+						ids.GenerateTestShortID(),
+					},
+					ValidatorDeposits: [][]UTXODeposit{
+						{
+							{
+								depositOfferID,
+								10,
+								10,
+								"",
+							},
+						},
+					},
+					UTXODeposits: []UTXODeposit{
+						{
+							depositOfferID,
+							10,
+							10,
+							"",
+						},
+					},
+					DepositOffers: depositOffer,
+				},
+				Time:          5,
+				InitialSupply: 0,
+				Message:       "",
+				Encoding:      formatting.Hex,
+			},
+			reply:       BuildGenesisReply{},
+			expectedErr: errUTXOHasNoValue,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := buildCaminoGenesis(&tt.args, &tt.reply)
+			require.ErrorIs(t, err, tt.expectedErr)
+			if err != nil {
+				require.Equal(t, tt.reply.Bytes, "")
+				require.Equal(t, tt.reply.Encoding, formatting.Hex)
+			} else {
+				require.NotEqualValues(t, tt.reply.Bytes, "")
```

Comment:
well, it doesn't tell us if this reply bytes is right or not
u should compare it with actual expected value, otherwise some bullshit bytes like []byte("bullshit") will pass the test as well :)